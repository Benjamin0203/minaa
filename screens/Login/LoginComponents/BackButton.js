import React from 'react'
import { IconButton, Text, View } from 'react-native-paper'
import { TouchableOpacity, StyleSheet } from 'react-native'
import { getStatusBarHeight } from 'react-native-status-bar-height'

export default function BackButton({ goBack }) {
  return (
    <View
      style={styles.container}
    >
      <IconButton
        icon="arrow-left"
        size={30}
        onPress={goBack}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: getStatusBarHeight(),
    left: 4,
  },
})