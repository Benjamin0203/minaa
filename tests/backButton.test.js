import React from 'react';
import renderer from 'react-test-renderer';
import BackButton from '../screens/Login/LoginComponents/BackButton';
import { getStatusBarHeight } from 'react-native-status-bar-height'

const style =  {
  position: 'absolute',
  top: getStatusBarHeight(),
  left: 4,
}

describe('<BackButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BackButton style={style} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});