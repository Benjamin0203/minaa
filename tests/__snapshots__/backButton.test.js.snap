// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "end": undefined,
      "left": 4,
      "position": "absolute",
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "start": undefined,
      "top": 20,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 23,
          "borderWidth": 0,
          "elevation": 0,
          "height": 46,
          "margin": 6,
          "overflow": "hidden",
          "width": 46,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Array [
              Object {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 23,
              },
            ],
          ]
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
